src/ShallowTraversal.js

```diff
@@ -18,8 +21,17 @@ import {
 export function childrenOfNode(node) {
   if (!node) return [];
   const maybeArray = propsOfNode(node).children;
+  const recursiveWithout = array =>
+    map(without(array, false, undefined, null), value => {
+      if (isArray(value)) {
+        return recursiveWithout(value);
+      }
+      return value;
+    });
+
+  const filtered = isArray(maybeArray) ? recursiveWithout(maybeArray) : maybeArray;
   const result = [];
-  React.Children.forEach(maybeArray, child => result.push(child));
+  React.Children.forEach(filtered, child => result.push(child));
```

Comment:
a simpler solution here might be to just check and see if `child` is `false`, `null`, or `undefined` before pushing it into the result?  The recursive nature of children will be taken care of in `React.Children.forEach` in this case.